kong/client.go

```diff
@@ -107,6 +112,12 @@ type Status struct {
 	ConfigurationHash string `json:"configuration_hash,omitempty" yaml:"configuration_hash,omitempty"`
 }
 
+func WithDoer(doer Doer) Option {
```

Comment:
```suggestion
// WithDoer sets the Client's request dispatcher, the function that sends a request to Kong and returns its response.
func WithDoer(doer Doer) Option {
```

In addition to the comment, any reason not to make this a method? That's more typical for the various setters, such as `Client.SetWorkspace()`.

Also, since I can't comment on the unmodified line, please update the comment for `Client.Do()` to mention the override:

```

// Do executes an HTTP request and returns a Response.
//
// The caller can optionally provide v parameter, which when provided will contain
// the response body. Do supports wither an io.Writer (which will contain the
// response body verbatim) or anything else which the body should be unmarshalled
// into.
//
// By default, Do() calls DoRaw() to send the request and return the response before unmarshalling, logging,
// and error handling. The Client's WithDoer() method allows overriding this to inject custom behavior.
```